indico/modules/events/timetable/controllers/manage.py

```diff
@@ -113,3 +114,7 @@ def _process_PATCH(self):
         if updates:
             update_timetable_entry(self.timetable_entry, updates)
         return jsonify()
+
+    def _process_DELETE(self):
+        """Delete a timetable entry"""
+        delete_timetable_entry(self.timetable_entry)
```

Comment:
delete contributions if `self.event_new.type != 'conference'`; check in prod what we do for session blocks (deleting the last session block also deletes the session in case of a meeting for example)